### ISSUE TITLE
Enforce curly brackets position

### ISSUE THREAD

**Issue by ficristo** [eslint/eslint#6051]
I would like to have a rule similar to the one I've asked on JSCS https://github.com/jscs-dev/node-jscs/issues/1956.
I tried to implement it at https://github.com/jscs-dev/node-jscs/pull/2035, but there are still some problems https://github.com/jscs-dev/node-jscs/issues/2077.
I found some related rules: `curly` and `object-curly-spacing`, but none similar of what I would like to have.

**Comment by ficristo** [comment]
Actually could be an option for `brace-style` rule.

**Comment by eslintbot** [comment]
Hi @ficristo, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by ficristo** [comment]
ATM I don't use ESLint, but since JSCS is deprecated I would like to have this options in ESLint and switch to it.
I would like to enforce the `brace-style` and the options `1tbs` but with an exception.
On cases where the conditions span on multiple lines I would like to put the brackets on a new line.
Example with an `if`:
```javascript
// 1tbs
if (cond1 && cond2 &&
    cond3...) { // <-- brackets 

// 1tbs with the exception
if (cond1 && cond2 &&
    cond3...)
{ // <-- brackets on a new line
```
This should enforced on other cases:
```javascript
// 1tbs
for (inits;
     conds;
     steps) { // <-- brackets 

// 1tbs with the exception
for (inits;
     conds;
     steps)
{ // <-- brackets on a new line
```

**Comment by kaicataldo** [comment]
Thanks for adding more info and clarifying. Just to be clear, this is not a current feature in JSCS, right? This is an enhancement you're advocating for in both projects?

**Comment by ficristo** [comment]
Actually I've already implemented it in JSCS.
Unfortunately it has some bugs that make it unusable...
Since JSCS is deprecated I'm asking to add it to ESLint.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

**Comment by ficristo** [comment]
@alberto do you think is possible to achive this with a plugin?

**Comment by alberto** [comment]
@ficristo Yes, you can definitely create a custom rule adapted to your needs using `brace-style` as a starting point. 👍
